src/test/java/com/mysql/cj/jdbc/ha/plugins/AWSSecretsManagerPluginTest.java

```diff
@@ -0,0 +1,189 @@
+import static org.mockito.Mockito.when;
+
+public class AWSSecretsManagerPluginTest {
+
+  private static final String TEST_REGION = "us-east-2";
+  private static final String TEST_SECRET_ID = "secretId";
+  private static final String VALID_SECRET_STRING = "{\"username\": \"testUser\", \"password\": \"testPassword\"}";
+  private static final String INVALID_SECRET_STRING = "{username: invalid, password: invalid}";
+  private static final Pair<String, Region> SECRET_CACHE_KEY = Pair.of(TEST_SECRET_ID, Region.of(TEST_REGION));
+  private static final AWSSecretsManagerPlugin.Secret SECRET = new AWSSecretsManagerPlugin.Secret("testUser", "testPassword");
+  private static final String TEST_DB_URL = "jdbc:mysql:aws://test-domain:3306/test";
+  private static final String TEST_SQL_ERROR = "SQL exception error message";
+  private static final String SQL_STATE_FAIL_GENERIC = "HY000";
+  private static final GetSecretValueResponse VALID_GET_SECRET_VALUE_RESPONSE = GetSecretValueResponse.builder().secretString(VALID_SECRET_STRING).build();
+  private static final GetSecretValueResponse INVALID_GET_SECRET_VALUE_RESPONSE = GetSecretValueResponse.builder().secretString(INVALID_SECRET_STRING).build();
+  private final PropertySet propertySet = new JdbcPropertySetImpl();
+  private AWSSecretsManagerPlugin plugin;
+  private AutoCloseable closeable;
+  private ConnectionUrl connectionUrl;
+
+  @Mock ConnectionProxy proxy;
+  @Mock IConnectionPlugin nextPlugin;
+  @Mock Log logger;
+  @Mock SecretsManagerClient mockSecretsManagerClient;
+  @Mock GetSecretValueRequest mockGetValueRequest;
+  @Captor ArgumentCaptor<ConnectionUrl> captor;
+
+  @BeforeEach
+  private void init() throws SQLException {
+    closeable = MockitoAnnotations.openMocks(this);
+
+    Properties properties = new Properties();
+    properties.setProperty(AWSSecretsManagerPlugin.SECRET_ID_PROPERTY, TEST_SECRET_ID);
+    properties.setProperty(AWSSecretsManagerPlugin.REGION_PROPERTY, TEST_REGION);
+    this.propertySet.initializeProperties(properties);
+
+    this.connectionUrl = ConnectionUrl.getConnectionUrlInstance(TEST_DB_URL, properties);
+
+    this.plugin = new AWSSecretsManagerPlugin(
+        proxy,
+        propertySet,
+        nextPlugin,
+        logger,
+        mockSecretsManagerClient,
+        mockGetValueRequest);
+  }
+
+  @AfterEach
+  void cleanUp() throws Exception {
+    closeable.close();
+    AWSSecretsManagerPlugin.SECRET_CACHE.clear();
+  }
+
+  @Test
+  public void testConnectWithCachedSecrets() throws SQLException {
+    // Add initial cached secret to be used for a connection
+    AWSSecretsManagerPlugin.SECRET_CACHE.put(SECRET_CACHE_KEY, SECRET);
+
+    this.plugin.openInitialConnection(this.connectionUrl);
+
+    assertEquals(1, AWSSecretsManagerPlugin.SECRET_CACHE.size());
+    verify(this.mockSecretsManagerClient, never()).getSecretValue(this.mockGetValueRequest);
+    verify(this.nextPlugin).openInitialConnection(any());
```

Comment:
I believe you need to check properties inside a ConnectionUrl object (that has been passed to openInitialConnection) and check whether username and password are presented in those properties.